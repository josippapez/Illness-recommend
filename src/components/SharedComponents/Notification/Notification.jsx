import React, { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { ToastContainer, toast } from 'react-toastify';

const Notification = () => {
  const user = useSelector(state => state.user);
  const userInfo = useSelector(state => state.user.userInfo);
  const currentPatientInfo = useSelector(state => state.patient);
  const medicationInfo = useSelector(
    state => state.medicationList.medicationInfo
  );

  const statusArray = [user, userInfo, medicationInfo, currentPatientInfo];
  useEffect(() => {
    statusArray.map(statusInfo => {
      if (
        statusInfo &&
        (statusInfo.status === 200 || statusInfo.status === 201)
      ) {
        toast.info(statusInfo.message, {
          position: 'top-center',
          autoClose: 3000,
          hideProgressBar: false,
          closeOnClick: true,
          pauseOnHover: true,
          draggable: true,
          progress: undefined,
        });
      }
      if (
        statusInfo &&
        statusInfo.status &&
        statusInfo.status !== 200 &&
        statusInfo.status !== 201
      ) {
        toast.error(
          statusInfo.message ? statusInfo.message : 'Nešto je pošlo po krivu!',
          {
            position: 'bottom-center',
            autoClose: 3000,
            hideProgressBar: false,
            closeOnClick: true,
            pauseOnHover: true,
            draggable: true,
            progress: undefined,
          }
        );
      }
    });
  }, [statusArray]);

  return <ToastContainer />;
};

export default Notification;
