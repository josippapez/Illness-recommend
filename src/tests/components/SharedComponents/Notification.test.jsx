/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import Notification from '../../../components/SharedComponents/Notification/Notification';

const initState = {
  user: {
    id: 1,
    name: 'Anno Anno',
    email: 'test@tt.com',
    role: 'user',
    isAuthenticated: true,
    error: null,
    status: null,
    message: null,
    userInfo: {
      data: {
        id: null,
        name: null,
        email: null,
        role: null,
      },
      error: null,
      status: null,
    },
  },
  usersList: {
    users: null,
    error: null,
    status: null,
  },
  alergies: {
    alergies: [
      {
        id: 1,
        name: 'Amikacin',
      },
      {
        id: 2,
        name: 'Azitromicin',
      },
    ],
    error: null,
    status: null,
  },
  medicationList: {
    medicationInfo: {
      data: null,
      error: null,
      status: null,
    },
    medications: null,
    error: null,
    status: null,
  },
  symptoms: {
    symptoms: null,
    error: null,
    status: null,
    message: null,
  },
  theme: {
    darkTheme: false,
  },
};
const mockStore = configureStore();
const store = mockStore({
  ...initState,
  patient: {
    currentPatientInfo: null,
    viewingPatientInfo: null,
    error: null,
    status: 400,
    message: 'Podaci ne uspješno spremljeni',
    patientList: null,
  },
});

describe('Notification', () => {
  let component;

  it('Renders correctly when no request successfuly finished', () => {
    component = mount(
      <Provider store={store}>
        <Notification />
      </Provider>
    );
    expect(component.html()).toMatchSnapshot();
  });

  it('Renders correctly when request successfuly finished', () => {
    const mockInitState = {
      ...initState,
      patient: {
        currentPatientInfo: null,
        viewingPatientInfo: null,
        error: null,
        status: 201,
        message: 'Podaci uspješno spremljeni',
        patientList: null,
      },
    };
    const mockMockStore = configureStore();
    const mockStore = mockMockStore(mockInitState);
    component = mount(
      <Provider store={mockStore}>
        <Notification />
      </Provider>
    );
    expect(component.html()).toMatchSnapshot();
  });

  it('Modal closes on close button click', () => {
    const mockInitState = {
      ...initState,
      patient: {
        currentPatientInfo: null,
        viewingPatientInfo: null,
        error: null,
        status: 201,
        message: 'Podaci uspješno spremljeni',
        patientList: null,
      },
    };
    const mockMockStore = configureStore();
    const mockStore = mockMockStore(mockInitState);
    component = mount(
      <Provider store={mockStore}>
        <Notification />
      </Provider>
    );
    component.find('.Toastify').simulate('click');
    expect(component.html()).toMatchSnapshot();
  });
});
