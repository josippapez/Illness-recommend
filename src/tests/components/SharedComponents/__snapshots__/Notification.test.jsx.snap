// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Modal closes on close button click 1`] = `"<div class=\\"Toastify\\"></div>"`;

exports[`Notification Renders correctly when no request successfuly finished 1`] = `"<div class=\\"Toastify\\"></div>"`;

exports[`Notification Renders correctly when request successfuly finished 1`] = `"<div class=\\"Toastify\\"></div>"`;
